microsoft-edge/webview2/concepts/process-related-events.md

```diff
@@ -0,0 +1,190 @@
+---
+title: Handling Process-related Events in WebView2
+description: Process-related events in WebView2 and how apps can handle them for recovery and improved reliability.
+author: MSEdgeTeam
+ms.author: msedgedevrel
+ms.topic: conceptual
+ms.prod: microsoft-edge
+ms.technology: webview
+ms.date: 06/20/2023
+---
+# Handling Process-related Events in WebView2
+## Overview
+The WebView2 API provides the `CoreWebView2.ProcessFailed` and `CoreWebView2Environment.BrowserProcessExited` events for your application to react to different scenarios. Use this document to learn how to use these events to react when these scenarios present.
+
+  * `CoreWebView2.ProcessFailed`. Use this event for diagnostics and recovery from failures in the WebView2 processes.
+  * `CoreWebView2Environment.BrowserProcessExited`. Use this event to synchronize operations involving WebView2 Runtime resources and lifetime, such as User Data Folder management and updates.
+
+To improve the reliability of your WebView2 application, it is recommended that it handles at least the following events:
+  * [The main browser process has exited unexpectedly](<link>).
+  * [A renderer process has exited unexpectedly](<link>).
+  * [A renderer process has become unresponsive](<link>).
+
+> [!NOTE]
+> This document is a high-level overview of the aforementioned events. Ultimately, the reference documentation for each individual API mentioned in this document shall prevail.
+
+## WebView2 Process Model and related APIs
+The [WebView2 Process Model](process-model.md) uses the WebView2 Runtime to support your WebView2 application. Because most of the magic in the WebView2 controls in your application happens out-of-process (in the WebView2 Runtime), failures and general information about these processes are passed to your application through different WebView2 APIs. The APIs used to report this information to your application include:
+
+  * CoreWebView2
+    - `BrowserProcessId` property
+    - `ProcessFailed` event
+  * CoreWebView2Environment
+    - `BrowserProcessExited` event
+    - `ProcessInfosChanged` event
+    - `GetProcessInfos` method
+
```

Comment:
```suggestion
##### [.NET/C#](#tab/dotnetcsharp)

* `CoreWebView2` Class:
   * [CoreWebView2.BrowserProcessId Property](/dotnet/api/microsoft.web.webview2.core.corewebview2.browserprocessid)
   * [CoreWebView2.ProcessFailed Event](/dotnet/api/microsoft.web.webview2.core.corewebview2.processfailed)
* `CoreWebView2Environment` Class:
   * [CoreWebView2Environment.BrowserProcessExited Event](/dotnet/api/microsoft.web.webview2.core.corewebview2environment.browserprocessexited)
   * [CoreWebView2Environment.ProcessInfosChanged Event](/dotnet/api/microsoft.web.webview2.core.corewebview2environment.processinfoschanged)
   * [CoreWebView2Environment.GetProcessInfos Method](/dotnet/api/microsoft.web.webview2.core.corewebview2environment.getprocessinfos)

##### [WinRT/C#](#tab/winrtcsharp)

* `CoreWebView2` Class:
   * [CoreWebView2.BrowserProcessId Property](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2#browserprocessid)
   * [CoreWebView2.ProcessFailed Event](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2#processfailed)
* `CoreWebView2Environment` Class:
   * [CoreWebView2Environment.BrowserProcessExited Event](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2environment#browserprocessexited)
   * [CoreWebView2Environment.ProcessInfosChanged Event](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2environment#processinfoschanged)
   * [CoreWebView2Environment.GetProcessInfos Method](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2environment#getprocessinfos)

##### [Win32/C++](#tab/win32cpp)

* `ICoreWebView2` interface:
   * [ICoreWebView2::get_BrowserProcessId method](/microsoft-edge/webview2/reference/win32/icorewebview2#get_browserprocessid)<!--no put-->
   * [ICoreWebView2::add_ProcessFailed method](/microsoft-edge/webview2/reference/win32/icorewebview2#add_processfailed)
   * [ICoreWebView2::remove_ProcessFailed method](/microsoft-edge/webview2/reference/win32/icorewebview2#remove_processfailed)
* [ICoreWebView2Environment5 interface](/microsoft-edge/webview2/reference/win32/icorewebview2environment5#add_browserprocessexited)
   * [ICoreWebView2Environment5::add_BrowserProcessExited event](/microsoft-edge/webview2/reference/win32/icorewebview2environment5#add_browserprocessexited)
   * [ICoreWebView2Environment5::remove_BrowserProcessExited event](/microsoft-edge/webview2/reference/win32/icorewebview2environment5#remove_browserprocessexited)
* [ICoreWebView2Environment8 interface](/microsoft-edge/webview2/reference/win32/icorewebview2environment8)
   * [ICoreWebView2Environment8::add_ProcessInfosChanged event](/microsoft-edge/webview2/reference/win32/icorewebview2environment8#add_processinfoschanged)
   * [ICoreWebView2Environment8::remove_ProcessInfosChanged event](/microsoft-edge/webview2/reference/win32/icorewebview2environment8#remove_processinfoschanged)
   * [ICoreWebView2Environment8::GetProcessInfos method](/microsoft-edge/webview2/reference/win32/icorewebview2environment8#getprocessinfos)

---

```
curious to see the non-linked, easy-to-read list above, with this working tab-set below it (how readable/comprehensible?)